hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -503,6 +505,56 @@ public void storeCertificate(String pemEncodedCert,
         certificateCodec, true);
   }
 
+  /**
+   * Stores all Certificates in this response on local disk.
+   * @param response - response from SCM
+   * @return first certificate ID
+   * @throws CertificateException - on Error
+   */
+  public String storeCertificate(SCMGetCertResponseProto response)
+      throws CertificateException {
+    return storeCertificate(response,
+        getSecurityConfig().getCertificateLocation(getComponentName()));
+  }
+
+  protected String storeCertificate(SCMGetCertResponseProto response,
+      Path codecPath) throws CertificateException {
+    CertificateCodec certCodec = new CertificateCodec(securityConfig,
+        codecPath);
+    try {
+      String pemEncodedCert = response.getX509Certificate();
+      if (!response.hasX509RootCACertificate()) {
+        throw new CertificateException("Unable to find SCM root certificate");
+      }
+      String pemEncodedRootCert = response.getX509RootCACertificate();
+      X509Certificate rootCert =
+          CertificateCodec.getX509Certificate(pemEncodedRootCert);
+      CertPath path =
+          CertificateCodec.getCertPathFromPemEncodedString(pemEncodedCert);
+      for (Certificate c: path.getCertificates()) {
+        X509Certificate cert = (X509Certificate) c;
+        CAType caType = CAType.NONE;
+        if (cert.getBasicConstraints() != -1) {
+          if (rootCert.getSerialNumber().toString().equals(
+              cert.getSerialNumber().toString())) {
+            caType = CAType.ROOT;
+          } else {
+            caType = CAType.SUBORDINATE;
+          }
+        }
+        String certName = String.format(CERT_FILE_NAME_FORMAT,
+            caType.getFileNamePrefix() + cert.getSerialNumber().toString());
+        certCodec.writeCertificate(
+            CertificateCodec.getCertificateHolder(cert), certName);
+      }
+      return ((X509Certificate)path.getCertificates().get(0))
```

Comment:
Q. is the first certificate in the list the "default" certificate?  similar to :

>  // Return the default certificate ID

Reply:
the certificate path contains the certificate of the service itself as the first item followed by the chain of signer certificates up until the trust anchor (rootCA certificate), so in this case this will return the dn/om/scm/recon certificate's ID we got back with the certificate in the certificate bundle as a result of the getCert request.

Reply:
Makes sense.  Thanks.